mantle/kola/tests/rhcos/disk.go

```diff
@@ -0,0 +1,148 @@
+	register.RegisterTest(&register.Test{
+		Run:         varLibContainers,
+		ClusterSize: 0,
+		Name:        `rhcos.disk.varlibcontainers`,
+		Flags:       []register.Flag{},
+		Distros:     []string{"rhcos"},
+		Platforms:   []string{"qemu-unpriv"},
+	})
+}
+
+func varLibContainers(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	disk := "secondary"
+	varMountName := "var-lib-containers.mount"
+	mkfsServiceName := fmt.Sprintf("systemd-mkfs@dev-disk-by-id-virtio-%s.service", disk)
+
+	options := platform.MachineOptions{
+		AdditionalDisks: []platform.Disk{
+			{Size: "1024M", DeviceOpts: []string{"serial=secondary"}},
+		},
+	}
+
+	mkfsUnit := fmt.Sprintf(`[Unit]
+	Description=Make File System on /dev/disk/by-id/%[1]s
+	DefaultDependencies=no
+	BindsTo=dev-disk-by\x2did-virtio\x2d%[1]s.device
+	After=dev-disk-by\x2did-virtio\x2d%[1]s.device var.mount
+	Before=systemd-fsck@dev-disk-by\x2did-virtio\x2d%[1]s.service
+	Before=shutdown.target
+
+	[Service]
+	Type=oneshot
+	RemainAfterExit=yes
+	ExecStart=/bin/bash -c "/bin/rm -rf /var/lib/containers/*"
+	ExecStart=mkfs.xfs /dev/disk/by-id/virtio-%[1]s
+	TimeoutSec=0
+
+	[Install]
+	WantedBy=var-lib-containers.mount`, disk)
+
+	varMountUnit := fmt.Sprintf(`[Unit]
+	Description=Mount secondary disk to /var/lib/containers
+	Before=local-fs.target
+	Requires=%[1]s
+	After=%[1]s
+
+	[Mount]
+	What=/dev/disk/by-id/virtio-%[2]s
+	Where=/var/lib/containers
+	Type=xfs
+	Options=defaults,prjquota
+
+	[Install]
+	WantedBy=local-fs.target`, mkfsServiceName, disk)
+
+	config := ignv3types.Config{
+		Ignition: ignv3types.Ignition{
+			Version: "3.0.0",
+		},
+		Systemd: ignv3types.Systemd{
+			Units: []ignv3types.Unit{
+				{
+					Name:     mkfsServiceName,
+					Contents: &mkfsUnit,
+					Enabled:  util.BoolToPtr(true),
+				},
+				{
+					Name:     varMountName,
+					Contents: &varMountUnit,
+					Enabled:  util.BoolToPtr(false),
+				},
+			},
+		},
+	}
+	convertedConf, err := ignconverter.Translate3to2(config)
+	buf, err := json.Marshal(convertedConf)
+	if err != nil {
+		c.Fatalf("Unable to Marshal config")
+	}
+	serializedConfig := buf
+
+	switch pc := c.Cluster.(type) {
+	// These cases have to be separated because when put together to the same case statement
+	// the golang compiler no longer checks that the individual types in the case have the
+	// NewMachineWithOptions function, but rather whether platform.Cluster does which fails
+	case *unprivqemu.Cluster:
+		m, err = pc.NewMachineWithOptions(conf.Ignition(string(serializedConfig)), options, true)
+	default:
+		c.Fatal("unknown cluster type")
+	}
+
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Verify /var/lib/containers isn't mounted then enable the units to move
+	// it to the secondary disk
+	vlc := "/var/lib/containers"
+	if mount, err := c.SSH(m, fmt.Sprintf("findmnt %s -n -o SOURCE", vlc)); err == nil {
+		c.Fatalf("%s is already mounted on %s", vlc, string(mount))
+	}
+	c.MustSSH(m, "podman run docker.io/hello-world")
```

Comment:
More pulling from docker hub is going to exacerbate our current CI issues with the hub rate limiting us.

Not something you need to solve in this PR necessarily, just noting this.

Reply:
I was originally trying to build a scratch container but then I needed to get a Dockerfile and a binary with no dependencies (and somewhere to house the source for the binary).  Pulling a small container from the hub just seemed a lot simpler for a simple check.  Now that I think more about it, I can probably just build an empty scratch container and `podman build` to  exercise the storage in /var/lib/containers.

Reply:
The test now builds a small scratch container to sanity check the underlying storage after changing the disk that /var/lib/containers is mounted on.